### PR TITLE
Fix ENTER/SPACEBAR on card with tapAction

### LINKED ISSUE #3196: [Accessibility] Card with tapAction cannot be executed thru ENTER/SPACEBAR
## Version

`master` branch.

## Describe the bug

On a card with `tapAction` set, after <kbd>TAB</kbd> into the card component (root container), pressing <kbd>ENTER</kbd> or <kbd>SPACEBAR</kbd> key should execute the card.

## Steps to reproduce

1. Talk to MockBot
1. Type `receiptcard`
1. Press <kbd>TAB</kbd> key to move the focus to the card container
1. Press <kbd>ENTER</kbd> or <kbd>SPACEBAR</kbd>

## Expected behavior

It should open a new window, as stated by the `receiptcard` command.

## Additional context

[Bug]

### PR DESCRIPTION
**Pull request by compulim**
Fixes #3196.

## Changelog Entry

-  Fixed [#3196](https://github.com/microsoft/BotFramework-WebChat/issues/3196). Cards with `tapAction` should be executable by <kbd>ENTER</kbd> or <kbd>SPACEBAR</kbd> key, by [@compulim](https://github.com/compulim) in PR [#3197](https://github.com/microsoft/BotFramework-WebChat/pull/3197)

## Description

`tapAction` on receipt card cannot be executed using keyboard.

## Specific Changes

- Added `onKeyPress` handler for `<AdaptiveCardRenderer`>, in addition to existing `onClick` handler
- Moved `/__tests__/html/serve.json` to `/serve-test.json`
- Fixed `NUnitTestReporter` error when no tests were run
- Fixed `<AdaptiveCardRenderer>` and `<ErrorBox>` for not showing up errors properly
- Fixed `runHook` may keep running hook indefinitely, if the hook errored out while running
- Added "finish key" to assist running test with `sendKeys` et al. locally in a browser

---

-  [x] Testing Added

### CODE REVIEW

File: __tests__/html/adaptiveCards.tapAction.html [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        expect,
+        host,
+        pageObjects,
+        shareObservable,
+        timeouts,
+        token,
+        updateIn
+      } = window.WebChatTest;
+
+      function stringToArrayBuffer(value) {
```
Comment (tdurnford) [inline __tests__/html/adaptiveCards.tapAction.html:24]: Should we put this in a test util directory?

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 2]
```diff
@@ -235,18 +235,46 @@ const AdaptiveCardRenderer = ({ actionPerformedClassName, adaptiveCard, disabled
 
   const disabled = disabledFromComposer || disabledFromProps;
 
-  const handleClick = useCallback(
-    ({ target }) => {
+  // TODO: [P2] We should consider using `adaptiveCard.selectAction` instead.
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:238]: Open an issue to track this

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 3]
```diff
@@ -235,18 +235,46 @@ const AdaptiveCardRenderer = ({ actionPerformedClassName, adaptiveCard, disabled
...
         // If the user is clicking on something that is already clickable, do not allow them to click the card.
         // E.g. a hero card can be tappable, and image and buttons inside the hero card can also be tappable.
-        if (typeof tabIndex !== 'number' || tabIndex < 0) {
-          tapAction && performCardAction(tapAction);
+        if (typeof tabIndex === 'number' && tabIndex >= 0) {
+          return;
         }
       }
+
+      performCardAction(tapAction);
```
Comment (tdurnford) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:276]: Do we need to make sure the tap action isn't undefined?

Comment (compulim) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:276]: Yes, it is checked in the `onClick` handler below.

I moved the check outside of `useCallback` to reduce the chance it get waste-render. I will put it side-by-side for easier reading.

**Pull request merged**